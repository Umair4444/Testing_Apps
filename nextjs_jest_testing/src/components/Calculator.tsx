"use client";
import { useState } from "react";

export default function Calculator() {
  const [num1, setNum1] = useState<number | string>("");
  const [num2, setNum2] = useState<number | string>("");
  const [result, setResult] = useState<number | null>(null);

  const calculate = (operator: string) => {
    const a = Number(num1);
    const b = Number(num2);
    if (isNaN(a) || isNaN(b)) {
      alert("Please enter valid numbers!");
      return;
    }

    let res = 0;
    switch (operator) {
      case "+":
        res = a + b;
        break;
      case "-":
        res = a - b;
        break;
      case "*":
        res = a * b;
        break;
      case "/":
        if (b === 0) {
          alert("Cannot divide by zero!");
          return;
        }
        res = a / b;
        break;
      default:
        return;
    }
    setResult(res);
  };

  return (
    <div className="max-w-sm mx-auto p-4 border rounded-lg shadow-md mt-10 text-center">
      <h1 className="text-2xl font-bold mb-4">ðŸ§® Simple Calculator</h1>
      <input
        type="number"
        value={num1}
        onChange={(e) => setNum1(e.target.value)}
        placeholder="Enter first number"
        className="w-full p-2 mb-2 border rounded"
      />
      <input
        type="number"
        value={num2}
        onChange={(e) => setNum2(e.target.value)}
        placeholder="Enter second number"
        className="w-full p-2 mb-4 border rounded"
      />

      <div className="flex justify-center gap-2 mb-4">
        <button
          onClick={() => calculate("+")}
          className="bg-blue-500 text-white px-4 py-2 rounded"
        >
          +
        </button>
        <button
          onClick={() => calculate("-")}
          className="bg-green-500 text-white px-4 py-2 rounded"
        >
          âˆ’
        </button>
        <button
          onClick={() => calculate("*")}
          className="bg-yellow-500 text-white px-4 py-2 rounded"
        >
          Ã—
        </button>
        <button
          onClick={() => calculate("/")}
          className="bg-red-500 text-white px-4 py-2 rounded"
        >
          Ã·
        </button>
      </div>

      {result !== null && (
        <div className="text-lg font-semibold">
          âœ… Result: <span className="font-bold">{result}</span>
        </div>
      )}
    </div>
  );
}
