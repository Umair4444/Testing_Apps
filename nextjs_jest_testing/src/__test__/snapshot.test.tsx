// import { render } from "@testing-library/react";
// import Calculator from "@/components/Calculator";

// describe("Calculator Snapshot Test", () => {
//   it("matches the snapshot", () => {
//     const { asFragment } = render(<Calculator />);
//     expect(asFragment()).toMatchSnapshot();
//   });
// });

import { render } from "@testing-library/react";
import Calculator from "@/components/Calculator";
import Greeting from "@/components/Greeting";

describe("UI Snapshot Test", () => {
  it("matches the snapshot of Calculator + Greeting", () => {
    const { asFragment } = render(
      <>
        <Calculator />
        <Greeting name="Leonardo" />
      </>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
