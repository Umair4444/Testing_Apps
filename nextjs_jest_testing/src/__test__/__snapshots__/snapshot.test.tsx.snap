// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`UI Snapshot Test matches the snapshot of Calculator + Greeting 1`] = `
<DocumentFragment>
  <div
    class="max-w-sm mx-auto p-4 border rounded-lg shadow-md mt-10 text-center"
  >
    <h1
      class="text-2xl font-bold mb-4"
    >
      ðŸ§® Simple Calculator
    </h1>
    <input
      class="w-full p-2 mb-2 border rounded"
      placeholder="Enter first number"
      type="number"
      value=""
    />
    <input
      class="w-full p-2 mb-4 border rounded"
      placeholder="Enter second number"
      type="number"
      value=""
    />
    <div
      class="flex justify-center gap-2 mb-4"
    >
      <button
        class="bg-blue-500 text-white px-4 py-2 rounded"
      >
        +
      </button>
      <button
        class="bg-green-500 text-white px-4 py-2 rounded"
      >
        âˆ’
      </button>
      <button
        class="bg-yellow-500 text-white px-4 py-2 rounded"
      >
        Ã—
      </button>
      <button
        class="bg-red-500 text-white px-4 py-2 rounded"
      >
        Ã·
      </button>
    </div>
  </div>
  <h1>
    Hello, Leonardo ðŸ‘‹
  </h1>
</DocumentFragment>
`;
